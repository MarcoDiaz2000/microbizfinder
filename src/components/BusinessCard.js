import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const BusinessCard = ({ business }) => (
  <div
    className="business-card"
    data-testid="business-card"
    style={{ backgroundImage: `url(${business.image_url})` }}
  >
    <h2>{business.name}</h2>
    <p>{business.categories.map((category) => category.title).join(', ')}</p>
    <p>{`Rating: ${business.rating} / 5`}</p>
    <Link to={`/details/${business.id}`}>See details</Link>
  </div>
);

BusinessCard.propTypes = {
  business: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    image_url: PropTypes.string,
    categories: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string,
    })),
    location: PropTypes.shape({
      address1: PropTypes.string,
      city: PropTypes.string,
    }),
    rating: PropTypes.number,
  }).isRequired,
};

export default BusinessCard;
