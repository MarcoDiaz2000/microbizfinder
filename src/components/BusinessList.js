import React from 'react';
import PropTypes from 'prop-types';
import BusinessCard from './BusinessCard';

const BusinessList = ({ businesses }) => {
  if (!businesses) {
    return null;
  }

  return (
    <div>
      {businesses.map((business) => (
        <BusinessCard key={business.id} business={business} />
      ))}
    </div>
  );
};

BusinessList.propTypes = {
  businesses: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      image_url: PropTypes.string,
      rating: PropTypes.number,
      review_count: PropTypes.number,
    }),
  ).isRequired,
};

export default BusinessList;
