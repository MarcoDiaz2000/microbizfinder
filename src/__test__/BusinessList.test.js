import React from 'react';
import { render } from '@testing-library/react';
import BusinessList from '../components/BusinessList';
import BusinessCard from '../components/BusinessCard';

jest.mock('../components/BusinessCard', () => jest.fn(() => null));

const mockBusinesses = [
  {
    id: '1', name: 'Negocio 1', image_url: 'https://ejemplo.com/imagen1.jpg', rating: 4, review_count: 200,
  },
  {
    id: '2', name: 'Negocio 2', image_url: 'https://ejemplo.com/imagen2.jpg', rating: 3.5, review_count: 150,
  },
];

describe('BusinessList', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = render(<BusinessList businesses={mockBusinesses} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    expect(BusinessCard).toHaveBeenCalledTimes(mockBusinesses.length);
  });

  mockBusinesses.forEach((business) => {
    it(`passes correct props to BusinessCard for business with id ${business.id}`, () => {
      expect(BusinessCard).toHaveBeenCalledWith({ business }, {});
    });
  });

  it('matches snapshot', () => {
    expect(renderedComponent.asFragment()).toMatchSnapshot();
  });
});
