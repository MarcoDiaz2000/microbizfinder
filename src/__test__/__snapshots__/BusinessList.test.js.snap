// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessList matches snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
